### ISSUE TITLE
Metrics recording function during optimization

### ISSUE THREAD

**Issue by theogf** [STOR-i/GaussianProcesses.jl#96]
Maybe I missed it. But is there some kind of function tracking the loglikelihood on the training set/a test set etc during optimization? To get some convergence plots?

Thanks

**Comment by maximerischard** [comment]
I haven't tried this before, but have a look at the `store_trace` and `show_trace` options in the Optim package: [here's the documentation](https://github.com/JuliaNLSolvers/Optim.jl/blob/2b6c6ed1e527d7ed8dfe509db29f7243ac283883/docs/src/user/config.md). The `optimize!` function takes keyword arguments that get passed to `Optim.optimize`, so it should be possible to do. Feel free to reopen this issue if you get stuck.
